### PR TITLE
Add real-er restart for web using webkit inspection protocol

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Send a webkit command to reload the page after a hot restart. Also use the webkit connection to tear down the browser on exit, and exit the tool on tab close.

### CODE REVIEW

File: packages/flutter_tools/lib/src/web/chrome.dart [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      final String windowsPrefix = _kWindowsPrefixes.firstWhere((String prefix) {
+        if (prefix == null) {
+          return false;
+        }
+        final String path = fs.path.join(prefix, _kWindowsExecutable);
+        return fs.file(path).existsSync();
+      }, orElse: () => '.');
+      final String path = fs.path.join(windowsPrefix, _kWindowsExecutable);
+      if (!fs.file(path).existsSync()) {
+        throwToolExit('Chrome executable not found at $_kMacOSExecutable');
```
Comment (blasten) [inline packages/flutter_tools/lib/src/web/chrome.dart]: nit: _kWindowsExecutable

**Review by blasten — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
I'm going to push some integration tests as well

**Comment by jonahwilliams** [comment]
Added a task/manifest for the devicelab

File: dev/devicelab/lib/tasks/web_dev_mode_tests.dart [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                final File appDartSource = file(path.join(
+                    _editedFlutterGalleryDir.path, 'lib/gallery/app.dart',
+                ));
+                appDartSource.writeAsStringSync(
+                    appDartSource.readAsStringSync().replaceFirst(
+                        "'Flutter Gallery'", "'Updated Flutter Gallery'",
+                    )
+                );
+                process.stdin.writeln('R');
+                ++hotRestartCount;
```
Comment (blasten) [inline dev/devicelab/lib/tasks/web_dev_mode_tests.dart]: nit: maybe this can be `hotRestarStarted` or do you plan to use the count later on?

Comment (jonahwilliams) [reply dev/devicelab/lib/tasks/web_dev_mode_tests.dart]: We'll use this later on for benchmarks similar to the other dev mode tests.

**Pull request merged**